src/fdmdv2_main.cpp

```diff
@@ -4094,8 +4094,6 @@ void txRxProcessing()
                 fprintf(stderr, "  nout: %d\n", nout);
             }
             ret = codec2_fifo_write(cbData->outfifo1, outsound_card, nout);
-            // should never fire as we check there is enough room before entering while loop
-            assert(ret != -1);
```

Comment:
I actually just noticed this on OSX as well (happens the instant I push PTT) and it appears it's related to sample rate mismatches. For instance, my Griffin iMic USB sound card seems to default to 48000Hz while my MacBook Pro's internal sound card defaults to 44100. Setting both to 44100 in the app causes the crash not to occur.

Anyway, I haven't dug too much into the Codec2 end of things but does removing the assertion result in other side effects?

Reply:
No it's just checking an assumption about how many samples are in a FIFO.  The current Windows Beta has been built off this branch, and have been working OK for a several weeks.

Reply:
Cool. I'm fine with this fix, then. I'll see about building a new OSX image with it as well.